opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -229,115 +287,127 @@ impl<R: RuntimeChannel> LogProcessor for BatchLogProcessor<R> {
-                            config.max_export_timeout,
-                            exporter.as_mut(),
-                            &timeout_runtime,
-                            logs.split_off(0),
-                        )
-                        .await;
-
-                        if let Some(channel) = res_channel {
-                            if let Err(send_error) = channel.send(result) {
-                                otel_debug!(
-                                    name: "BatchLogProcessor.Flush.SendResultError",
-                                    error = format!("{:?}", send_error),
-                                );
-                            }
-                        }
+                    Ok(BatchMessage::ForceFlush(sender)) => {
+                        let result = export_with_timeout_sync(remaining_time, exporter.as_mut(), logs.split_off(0), &mut last_export_time);
+                        let _ = sender.send(result);
                     }
-                    // Stream has terminated or processor is shutdown, return to finish execution.
-                    BatchMessage::Shutdown(ch) => {
-                        let result = export_with_timeout(
-                            config.max_export_timeout,
-                            exporter.as_mut(),
-                            &timeout_runtime,
-                            logs.split_off(0),
-                        )
-                        .await;
-
-                        exporter.shutdown();
-
-                        if let Err(send_error) = ch.send(result) {
-                            otel_debug!(
-                                name: "BatchLogProcessor.Shutdown.SendResultError",
-                                error = format!("{:?}", send_error),
-                            );
-                        }
+                    Ok(BatchMessage::Shutdown(sender)) => {
+                        let result = export_with_timeout_sync(remaining_time, exporter.as_mut(), logs.split_off(0), &mut last_export_time);
+                        let _ = sender.send(result);
+
+                        //
+                        // break out the loop and return from the current background thread.
+                        //
                         break;
                     }
-                    // propagate the resource
-                    BatchMessage::SetResource(resource) => {
+                    Ok(BatchMessage::SetResource(resource)) => {
                         exporter.set_resource(&resource);
                     }
+                    Err(RecvTimeoutError::Timeout) => {
+                        // FIXME handle result
+                        let _result = export_with_timeout_sync(remaining_time, exporter.as_mut(), logs.split_off(0), &mut last_export_time);
+                    }
+                    Err(err) => {
+                        otel_error!(
+                            name: "BatchLogProcessor.ReceiveError",
```

Comment:
For high severity logs like this, we need to make them user friendly. Cannot use ReceiveError as it won't tell anything to end users.

Reply:
Is there a category for internal or unknown error?

Reply:
Changed it to internal error for now.